blocks/colour.js

```diff
@@ -34,100 +39,92 @@ goog.require('Blockly.Blocks');
-        .appendField(Blockly.Msg.COLOUR_RGB_RED);
-    this.appendValueInput('GREEN')
-        .setCheck('Number')
-        .setAlign(Blockly.ALIGN_RIGHT)
-        .appendField(Blockly.Msg.COLOUR_RGB_GREEN);
-    this.appendValueInput('BLUE')
-        .setCheck('Number')
-        .setAlign(Blockly.ALIGN_RIGHT)
-        .appendField(Blockly.Msg.COLOUR_RGB_BLUE);
-    this.setOutput(true, 'Colour');
-    this.setTooltip(Blockly.Msg.COLOUR_RGB_TOOLTIP);
-  }
-};
+  // Block for composing a colour from RGB components.
+  {
+    "type": "colour_rgb",
+    "message0": "%{BKY_COLOUR_RGB_TITLE} %{BKY_COLOUR_RGB_RED} %1 %{BKY_COLOUR_RGB_GREEN} %2 %{BKY_COLOUR_RGB_BLUE} %3",
+    "args0": [
+      {
+        "type": "input_value",
+        "name": "RED",
+        "check": "Number",
+        "align": "RIGHT"
+      },
+      {
+        "type": "input_value",
+        "name": "GREEN",
+        "check": "Number",
+        "align": "RIGHT"
+      },
+      {
+        "type": "input_value",
+        "name": "BLUE",
+        "check": "Number",
+        "align": "RIGHT"
+      }
+    ],
+    "output": "Colour",
+    "colour": "%{BKY_COLOUR_HUE}",
+    "helpUrl": "%{BKY_COLOUR_RGB_HELPURL}",
+    "tooltip": "%{BKY_COLOUR_RGB_TOOLTIP}"
+  },
 
-Blockly.Blocks['colour_blend'] = {
-  /**
-   * Block for blending two colours together.
-   * @this Blockly.Block
-   */
-  init: function() {
-    this.setHelpUrl(Blockly.Msg.COLOUR_BLEND_HELPURL);
-    this.setColour(Blockly.Blocks.colour.HUE);
-    this.appendValueInput('COLOUR1')
-        .setCheck('Colour')
-        .setAlign(Blockly.ALIGN_RIGHT)
-        .appendField(Blockly.Msg.COLOUR_BLEND_TITLE)
-        .appendField(Blockly.Msg.COLOUR_BLEND_COLOUR1);
-    this.appendValueInput('COLOUR2')
-        .setCheck('Colour')
-        .setAlign(Blockly.ALIGN_RIGHT)
-        .appendField(Blockly.Msg.COLOUR_BLEND_COLOUR2);
-    this.appendValueInput('RATIO')
-        .setCheck('Number')
-        .setAlign(Blockly.ALIGN_RIGHT)
-        .appendField(Blockly.Msg.COLOUR_BLEND_RATIO);
-    this.setOutput(true, 'Colour');
-    this.setTooltip(Blockly.Msg.COLOUR_BLEND_TOOLTIP);
+  // Block for blending two colours together.
+  {
+    "type": "colour_blend",
+    "message0": "%{BKY_COLOUR_BLEND_TITLE} %{BKY_COLOUR_BLEND_COLOUR1} %1 %{BKY_COLOUR_BLEND_COLOUR2} %2 %{BKY_COLOUR_BLEND_RATIO} %3",
+    "args0": [
+      {
+        "type": "input_value",
+        "name": "COLOUR1",
+        "check": "Colour",
+        "align": "RIGHT"
+      },
+      {
+        "type": "input_value",
+        "name": "COLOUR2",
+        "check": "Colour",
+        "align": "RIGHT"
+      },
+      {
+        "type": "input_value",
+        "name": "RATIO",
+        "check": "Number",
+        "align": "RIGHT"
+      }
+    ],
+    "output": "Colour",
+    "colour": "%{BKY_COLOUR_HUE}",
+    "helpUrl": "%{BKY_COLOUR_BLEND_HELPURL}",
+    "tooltip": "%{BKY_COLOUR_BLEND_TOOLTIP}"
   }
-};
+]);  // End defineBlocksWithJsonArray(...)
```

Comment:
Is this line now required?  If so, add a comment and also use it in blocks/math.js